src/components/ArticlePage.jsx

```diff
@@ -2,20 +2,47 @@ import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import { fetchArticle } from "../requests";
 import { Comments } from "./Comments";
+import { patchArticleLike } from "../requests";
 
 export function ArticlePage() {
   const { article_id } = useParams();
 
   const [articleData, setArticleData] = useState({});
   const [isLoading, setIsLoading] = useState(true);
+  const [likes, setLikes] = useState(0);
+  const [canLike, setCanlike] = useState(true);
+  const [likeText, setLikeText] = useState("Like");
 
   useEffect(() => {
     fetchArticle(article_id).then(({ data: { articles } }) => {
       setArticleData(articles);
       setIsLoading(false);
+      setLikes(articles.votes);
     });
   }, []);
 
+  const handleLike = () => {
+    if (canLike === true) {
+      setCanlike(false);
+      setLikes(likes + 1);
+      setLikeText("Liked");
+      patchArticleLike(article_id, { inc_votes: 1 }).catch((err) => {
+        if (err) {
+          setLikeText("Can't like post at this moment");
```

Comment:
This is great that you have handled the failure of the patch and fed back to the user. The increased number of votes is still included though along with the message which is not the best UX.